### PR TITLE
Add PigZombieMock

### PR DESCRIPTION
**Pull request by Nirbose**
# Description
PigZombie entity added
Modified WorldMock for entity spawn

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Code follows existing style.
- [x] Unit tests added (if applicable).

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/entity/PigZombieMock.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	}
+
+	@Override
+	public int getAnger()
+	{
+		return this.anger;
+	}
+
+	@Override
+	public void setAnger(int i)
```
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PigZombieMock.java]:
```suggestion
	public void setAnger(int level)
```

File: src/main/java/be/seeseemelk/mockbukkit/entity/PigZombieMock.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	{
+		this.anger = i;
+	}
+
+	@Override
+	public void setAngry(boolean b)
```
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PigZombieMock.java]:
```suggestion
	public void setAngry(boolean angry)
```

File: src/main/java/be/seeseemelk/mockbukkit/entity/PigZombieMock.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	{
+		this.angry = b;
```
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PigZombieMock.java]:
```suggestion
		this.setAnger(angry ? 400 : 0);
```

File: src/main/java/be/seeseemelk/mockbukkit/entity/PigZombieMock.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	}
+
+	@Override
+	public boolean isAngry()
+	{
+		return this.angry;
```
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PigZombieMock.java]:
```suggestion
		return this.getAnger() > 0;
```

**Review by Thorinwasher — COMMENTED** [review]
Use descriptive names. Also found a difference compared to paper implementation. Looks good otherwise

File: src/main/java/be/seeseemelk/mockbukkit/entity/PigZombieMock.java (hunk 4 above)
Comment (Nirbose) [reply src/main/java/be/seeseemelk/mockbukkit/entity/PigZombieMock.java]: Commit 
b215c30

File: src/main/java/be/seeseemelk/mockbukkit/entity/PigZombieMock.java (hunk 3 above)
Comment (Nirbose) [reply src/main/java/be/seeseemelk/mockbukkit/entity/PigZombieMock.java]: Commit 
b215c30

File: src/main/java/be/seeseemelk/mockbukkit/entity/PigZombieMock.java (hunk 2 above)
Comment (Nirbose) [reply src/main/java/be/seeseemelk/mockbukkit/entity/PigZombieMock.java]: Commit 
b215c30

File: src/main/java/be/seeseemelk/mockbukkit/entity/PigZombieMock.java (hunk 1 above)
Comment (Nirbose) [reply src/main/java/be/seeseemelk/mockbukkit/entity/PigZombieMock.java]: Commit 
b215c30

File: src/main/java/be/seeseemelk/mockbukkit/entity/PigZombieMock.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+import org.bukkit.entity.PigZombie;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.UUID;
+
+public class PigZombieMock extends ZombieMock implements PigZombie
+{
+
+	private int anger = 0;
+	private boolean angry = false;
```
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PigZombieMock.java]: This field is unused

**Review by Thorinwasher — APPROVED** [review]
Looks good to me

**Pull request merged**